compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1906,6 +1925,11 @@ private Result seekMatchInExpressionRhs(int lookahead, int currentDepth, int cur
                         ParserRuleContext.XML_NAVIGATE_EXPR, ParserRuleContext.MEMBER_ACCESS_KEY_EXPR,
                         ParserRuleContext.LET_CLAUSE_END };
                 break;
+            case ORDER_KEY:
+                alternatives = new ParserRuleContext[] { ParserRuleContext.COMMA, ParserRuleContext.EXPRESSION,
+                        ParserRuleContext.ASCENDING_KEYWORD, ParserRuleContext.DESCENDING_KEYWORD,
+                        ParserRuleContext.ORDER_CLAUSE_END };
+                break;
```

Comment:
Here we states which tokens can occur after some expression. 
eg: `... order by foo <cursor>`

since what follows after `by` keyword is an expression, we can put many tokens inplace of `<cursor>`, such as a binary operator (foo + bar), a dot (foo.bar), etc. So we need to add all those.

You can check for the common things compared to the other cases in the switch-case. Additionally, there can be `ParserRuleContext.ORDER_CLAUSE_END`.